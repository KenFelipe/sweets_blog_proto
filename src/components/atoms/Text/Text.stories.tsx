import { Story, Meta } from '@storybook/react'
import { Text, TextProps } from './Text'

export default {
  title: 'Atoms/Text',
  component: Text,
} as Meta

const Template: Story<TextProps> = args => <Text {...args} />

export const Primary = Template.bind({})
Primary.storyName = 'Primary'

Primary.args = {
  children: 'TextComponent',
}
