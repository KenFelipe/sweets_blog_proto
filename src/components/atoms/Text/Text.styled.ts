import styled, { css } from 'styled-components'

export const Container = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.size.secondary};
    color: ${theme.color.secondary};
  `}
`
