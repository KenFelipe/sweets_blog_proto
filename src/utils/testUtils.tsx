import { render, RenderOptions } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'

import { theme } from '../styles/theme'

type Props = {
  children: React.ReactNode
}

const AllTheProviders: React.FC = ({ children }: Props) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
) => render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'

export { customRender as render }
