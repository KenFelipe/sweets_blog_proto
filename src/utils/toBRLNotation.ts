export function toBRLNotation(value: number) {
  return 'R$ ' + value.toFixed(2).replace('.', ',')
}
