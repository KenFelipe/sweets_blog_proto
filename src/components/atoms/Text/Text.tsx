import * as Styled from './Text.styled'

export type TextProps = {
  children: React.ReactNode
}

export const Text = ({ children }: TextProps) => {
  return <Styled.Container>{children}</Styled.Container>
}
