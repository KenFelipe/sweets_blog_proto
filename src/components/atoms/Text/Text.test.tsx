import { render, screen } from '@/utils/testUtils'
import { Text } from './Text'

it('render label correctly', () => {
  const { container } = render(<Text>react-testing-library</Text>)

  expect(screen.getByText(/react-testing-library/)).toBeInTheDocument()
  expect(container).toMatchSnapshot()
})
