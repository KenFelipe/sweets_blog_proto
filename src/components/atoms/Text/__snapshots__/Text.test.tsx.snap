// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render label correctly 1`] = `
.c0 {
  font-size: 3.2rem;
  color: pink;
}

<div>
  <p
    class="c0"
  >
    react-testing-library
  </p>
</div>
`;
